idseq_dag/__main__.py

```diff
@@ -34,6 +46,81 @@ def main():
     log.write("start executing the dag")
     flow.start()
 
+def count_input_reads(input_files, max_fragments):
+    input_read_count = idseq_dag.util.count.reads_in_group(input_files[0],
+                                                           max_fragments=max_fragments)
+    counts_dict = dict(fastqs=input_read_count)
+    if input_read_count == len(input_files) * max_fragments:
+        counts_dict["truncated"] = input_read_count
+    with open("fastqs.count", "w") as count_file:
+        json.dump(counts_dict, count_file)
+
+def run_step():
+    root_logger = logging.getLogger()
+    root_logger.setLevel(level=logging.INFO)
+    stream_handler = logging.StreamHandler()
+    stream_handler.setFormatter(idseq_dag.util.log.JsonFormatter())
+    root_logger.addHandler(stream_handler)
```

Comment:
Is there any special reason why we are not using `idseq_dag.util.log` module? I don't really love that module, but considering the steps are all using it, maybe it should have be initialized as in:
```
import idseq_dag.util.log as log
log.configure_logger()
```

Reply:
Yes, that module misconfigures logging in a way that breaks stdout/stderr streams. In fact I am quite unhappy with the logic in that module, and want it retired completely.

Reply:
I wanted to dismantle that module many times, specially that bizarre lock that is there to prevent multithread locks, but nobody really knows how or if it really works at all. I support retiring it, or at least the parts that are based on black magic :-), because I have never been fully convinced those locks work for the purpose they have been created.